apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -912,14 +916,33 @@ impl FetchDependencyGraph {
     }
 
     fn remove_node(&mut self, node_index: NodeIndex) {
+        self.on_modification();
         let edges_to_remove: Vec<EdgeIndex> =
             self.graph.edges(node_index).map(|edge| edge.id()).collect();
         for edge in edges_to_remove {
             self.graph.remove_edge(edge);
         }
+        // Note: Probably the `remove_edge` calls are unnecessary, as `remove_node` should remove
```

Comment:
The API docs say that `remove_node()` should remove edges, so you can indeed remove the calls above to `remove_edge()`.